### PR TITLE
Update pickdb tasks to update embedded/application.properties

### PR DESCRIPTION
**Pull request by labkey-ankurj**
#### Rationale
This PR updates the gradle tasks like gradle build, pickdb and may be others to work with spring boot.

#### Related Pull Requests
* https://github.com/LabKey/server/pull/21

### CODE REVIEW

**Review by labkey-susanh — APPROVED** [review]
@labkey-ankurj I'll approve this since you cannot, but can you please review my changes in this branch?

File: src/main/groovy/org/labkey/gradle/plugin/Tomcat.groovy [hunk 1]
```diff
@@ -65,6 +74,62 @@ class Tomcat implements Plugin<Project>
...
+                        File logFile = new File(ServerDeployExtension.getEmbeddedServerDeployDirectory(project), EMBEDDED_LOG_FILE_NAME)
+                        if (!logFile.getParentFile().exists())
+                            logFile.getParentFile().mkdirs()
+                        if (!logFile.exists())
+                            logFile.createNewFile()
+                        FileOutputStream outputStream = new FileOutputStream(logFile)
+                        def env = []
+                        env += "PATH=${ServerDeployExtension.getEmbeddedServerDeployDirectory(project)}/bin${File.pathSeparator}${System.getenv("PATH")}"
+                        if (TeamCityExtension.isOnTeamCity(project))
+                            env +="R_LIBS_USER=${System.getenv("R_LIBS_USER") != null ? System.getenv("R_LIBS_USER") : project.rootProject.file("sampledata/rlabkey")}"
```
Comment (labkey-tchad) [inline src/main/groovy/org/labkey/gradle/plugin/Tomcat.groovy]: We should pass through `R_LIBS_USER` regardless of running on TeamCity. Also, we don't need to default to '`sampledata/rlabkey`' anymore.
```suggestion
                        if (System.getenv("R_LIBS_USER") != null)
                            env +="R_LIBS_USER=${System.getenv("R_LIBS_USER")}"
```

File: src/main/groovy/org/labkey/gradle/plugin/Tomcat.groovy [hunk 2]
```diff
@@ -65,6 +74,62 @@ class Tomcat implements Plugin<Project>
...
+                    def command = "curl -X POST $endpoint"
+                    task.logger.quiet("Sending command to $endpoint")
+                    def proc = command.execute()
+                    proc.waitFor()
+                    if (proc.exitValue() != 0)
+                        task.logger.warn("Shutdown command exited with non-zero status ${proc.exitValue()}.")
+                    else
+                        task.logger.quiet("Shutdown successful")
+                }
+        }
```
Comment (labkey-tchad) [inline src/main/groovy/org/labkey/gradle/plugin/Tomcat.groovy:69]: Will this work on Windows?

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy [hunk 3]
```diff
@@ -153,15 +171,17 @@ class ModuleDistribution extends DefaultTask
...
         {
             zipArchives()
+            if (shouldBuildEmbeddedArchive())
+                embeddedTomcatZipArchive()
         }
-        if (includeWarArchive)
-        {
-            warArchives()
-        }
+
```
Comment (labkey-tchad) [inline src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy]: Should we allow a distribution to have only the embedded archive?
```suggestion

        if (!includeZipArchive && !includeTarGZArchive && shouldBuildEmbeddedArchive())
        {
            embeddedTomcatTarArchive()
        }
```

**Review by labkey-tchad — APPROVED** [review]
Do we want to match our Java styling for curly brackets (new line for open brackets)? I noticedf some places that were changed that way but some new code has it the old way (no new line before open brackets).

File: src/main/groovy/org/labkey/gradle/plugin/Tomcat.groovy (hunk 2 above)
Comment (labkey-susanh) [reply src/main/groovy/org/labkey/gradle/plugin/Tomcat.groovy:69]: It should if the `curl` command is available.

File: src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy (hunk 3 above)
Comment (labkey-susanh) [reply src/main/groovy/org/labkey/gradle/task/ModuleDistribution.groovy]: Possibly, but the choice of Tar archive here is sort of arbitrary (although I also wonder if we still need to support both tar and zip archives, but that's for a different disucssion).   The `includeEmbeddedArchive` property could be a string property where we provide 'tar.gz', 'zip' or both ('tar.gz,zip').  Perhaps that's better.

**Review by labkey-susanh — COMMENTED** [review]
> Do we want to match our Java styling for curly brackets (new line for open brackets)? I noticedf some places that were changed that way but some new code has it the old way (no new line before open brackets).

I had been using Java styling at least for functions and for if blocks and the like as that seemed the (reasonable) default for Groovy, but now IntelliJ seems to think this other styling is the default way to go.  Since the default stying for closures in Groovy, which are quite prevalent, is rather unclear to me (even from a personal preference standpoint), I just can't get very excited about it either way.

**Pull request merged**